engine/src/test/java/org/camunda/bpm/engine/test/dmn/deployment/DecisionDefinitionTest.java

```diff
@@ -0,0 +1,129 @@
+import org.camunda.bpm.engine.repository.DeploymentWithDefinitions;
+import org.camunda.bpm.engine.test.util.ProcessEngineBootstrapRule;
+import org.camunda.bpm.engine.test.util.ProcessEngineTestRule;
+import org.camunda.bpm.engine.test.util.ProvidedProcessEngineRule;
+import org.camunda.bpm.model.dmn.Dmn;
+import org.camunda.bpm.model.dmn.DmnModelInstance;
+import org.camunda.bpm.model.dmn.HitPolicy;
+import org.camunda.bpm.model.dmn.impl.DmnModelConstants;
+import org.camunda.bpm.model.dmn.instance.Decision;
+import org.camunda.bpm.model.dmn.instance.DecisionTable;
+import org.camunda.bpm.model.dmn.instance.Definitions;
+import org.camunda.bpm.model.dmn.instance.Input;
+import org.camunda.bpm.model.dmn.instance.InputExpression;
+import org.camunda.bpm.model.dmn.instance.Output;
+import org.camunda.bpm.model.dmn.instance.Text;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+
+public class DecisionDefinitionTest {
+
+  protected ProcessEngineBootstrapRule bootstrapRule = new ProcessEngineBootstrapRule(configuration -> {
+    configuration.setHistoryTimeToLive("P30D");
+  });
+
+  protected ProvidedProcessEngineRule engineRule = new ProvidedProcessEngineRule(bootstrapRule);
+
+  protected ProcessEngineTestRule testRule = new ProcessEngineTestRule(engineRule);
+
+  @Rule
+  public RuleChain ruleChain = RuleChain.outerRule(bootstrapRule)
+      .around(engineRule)
+      .around(testRule);
+
+  protected RepositoryService repositoryService;
+
+  @Before
+  public void init() {
+    this.repositoryService = engineRule.getRepositoryService();
+  }
+
+  @Test
+  public void shouldUseHistoryTimeToLiveOnDecisionDefinitions() {
+    // given
+    DmnModelInstance model = createDmnModelInstance();
+    DeploymentBuilder builder = repositoryService.createDeployment().addModelInstance("foo.dmn", model);
+
+    // when
+    DeploymentWithDefinitions deployment = testRule.deploy(builder);
+
+    // then
+    assertThat(deployment.getDeployedDecisionDefinitions().size()).isEqualTo(1);
+    assertThat(deployment.getDeployedDecisionDefinitions().get(0).getHistoryTimeToLive()).isEqualTo(30);
+  }
+
+  protected DmnModelInstance createDmnModelInstance() {
+    return createDmnModelInstance(null);
+  }
+
+  protected DmnModelInstance createDmnModelInstance(Integer camundaHistoryTimeToLive) {
+    DmnModelInstance modelInstance = Dmn.createEmptyModel();
+    Definitions definitions = modelInstance.newInstance(Definitions.class);
+    definitions.setId(DmnModelConstants.DMN_ELEMENT_DEFINITIONS);
+    definitions.setName(DmnModelConstants.DMN_ELEMENT_DEFINITIONS);
+    definitions.setNamespace(DmnModelConstants.CAMUNDA_NS);
+    modelInstance.setDefinitions(definitions);
+
+
+    Decision decision = modelInstance.newInstance(Decision.class);
+    decision.setId("Decision-1");
+    decision.setName("foo");
+
+    if (camundaHistoryTimeToLive != null) {
```

Comment:
❓ I only see one call of this method which passes `null` for the method parameter. In which scenario is this condition `true`?

Reply:
Similar to the above point, the conditional can be removed. Or re-introduced in case another test is added which requires this parameter to be present.